02 markdown (#3)
carpentries-incubator/jekyll-pages-novice: _episodes/02-markdown.md

  - inline style
  - reference style
  - reference style makes it easier to re-use and refactor links
- link out to a more comprehensive guide, e.g. 
  - https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet
  - https://docs.gitlab.com/ee/user/markdown.html
- save to a file called index.md
- put directory under version control with git
  - link to SWC git-novice

> ## Reproduce Markdown Source Code: Heading, Text, Emphasis
>
> Try to reproduce the source code of this challenge from the following rendered view:
> 
> # This largest heading is followed
> ### by a two times smaller one.
>
> Vanilla text may contain *italic* and **bold words**.
>
> This **paragraph** is separated from the vanilla text paragraph by a blank line and  
> line breaks  
> are caused by two trailing spaces at the end of the lines.
> 
>
> > ## Solution
> > ~~~
> > # This largest heading is followed
> > ### by a two times smaller one.
> >
> > Vanilla text may contain *italic* and **bold words**.
> >
> > This **paragraph** is separated from the vanilla text paragraph by a blank line and  
> > these line breaks  
> > are caused by two trailing spaces at the end of the lines.
> > ~~~

> > > {: .markdown }

Reviewer: the lesson template (currently) has no highlighting rules for Markdown source. Best to change all these `{: .markdown }` tags to `{: .source }`

Author: See 5c040eb

> 
> New text blocks require **blank lines** and 
> 
> 23. enumerations
> 23. starting at any number
> 23. continue to increase
> 26. in steps of one.
>     1. Sub enumerations (in some editors have to start at **one**)
>     2. and can include
>        - sub lists
>     4. keeping the enumeration count.
>
> > ## Solution
> > ~~~
> > - A list of items
> > - can be
> >   - deeply
> >     - nested
> > - returning to the top-level like this.
> > 
> > New text blocks require **blank lines** and 
> > 
> > 23. enumerations
> > 23. starting at any number
> > 23. continue to increase
> > 26. in steps of one.
> >     1. Sub enumerations (in some editors have to start at **one**)
> >     2. and can include
> >        - sub lists
> >     4. **without** disturbing the enumeration count.
> > ~~~
> > {: .markdown }
> {: .solution }
{: .challenge }

> ## Add Code
> Fill in the blanks below to create this rendered view
> 
> Discussing code details can be very helpful.  
> Therefore, referring to variables, e.g. `path` in a formatted way is great.
>
> Larger code snippets look like this:
> ```
> print('Hello World')
> ```
>
> ~~~
> Discussing code details can be very helpful.  

> > Therefore, referring to variables, e.g. _path_ in a formatted way is great.

Reviewer: ```suggestion
> When doing so, it's great to be able to refer to variables, functions, etc, e.g. `path` in a formatted way.
```

>
> Larger code snippets look like this:
>
> ___
> msg = 'Hello World'
> print(msg)
> ___
> ~~~
> {: .source }
> 
> > ## Solution
> > ~~~
> > Discussing code details can be very helpful.  

> > > Therefore, referring to variables, e.g. `path` in a formatted way is great.

Reviewer: ```suggestion
> > When doing so, it's great to be able to refer to variables, functions, etc, e.g. `path` in a formatted way.
```

> >
> > Larger code snippets look like this:
> > ```
> > print('Hello World')
> > ```
> > ~~~
> > {: .markdown }
> {: .solution }
{: .challenge }

> ## Syntax Highlighting
> Syntax highlighting increases readability a lot.  

> > Guess in which programming language the following code snippets are written

Reviewer: ```suggestion
> Below are code snippets written in HTML, Python, and R.
> Guess in which programming language each snippet is written
```

I think the above makes the exercise easier for those coming to the lesson without direct experience of Python/R

> and fill the gaps to activate syntax highlighting creating this rendered view
> 
> -------------

> > ```R

Reviewer: ```suggestion
> ```r
```
code block highlighting tags should be lowercase

> print(paste("How","are","you?"), quote = FALSE)
> ```
> -------------
> 
> ```python
> s = "How are you?"
> print s
> ```
> -------------
> 
> ```html
> <!DOCTYPE html>
> <html>
> <body>
> > <a href=https://carpentries.org/>This is a link</a>
> > </body>
> </html>
> ```
> -------------
>
> ~~~
> ```_
> print(paste("How","are","you?"), quote = FALSE)
> ```
> 
> ```______
> s = "How are you?"
> print s
> ```
> 
> ```____
> <!DOCTYPE html>
> <html>
> <body>
> > <a href=https://carpentries.org/>This is a link</a>
> > </body>
> </html>
> ```
> ~~~
> {: .source }
>
> > ## Solution
> > ~~~
> > ```R
> > print(paste("How","are","you?"), quote = FALSE)
> > ```
> > 
> > ```python
> > s = "How are you?"
> > print s
> > ```
> > 
> > ```html
> > <!DOCTYPE html>
> > <html>
> > <body>
> > > <a href=https://carpentries.org/>This is a link</a>
> > > </body>
> > </html>
> ```
> > ~~~
> > {: .markdown }
> {: .solution }
{: .challenge }

> 

Reviewer: It would be great to add one more exercise about links here, perhaps a fill-in-the-blanks? No need to do that now, though, we can open an Issue/include a TODO to remind us later.

Author: Opened an [issue](https://github.com/tobyhodges/building-websites-with-jekyll-and-github-or-gitlab/issues/4#issue-668666187) and will resolve it.

> ## Flavors
>

> > Basic Markdown features are the same on all platforms such as Github and Gitlab.

Reviewer: ```suggestion
> Basic Markdown features are the same on all platforms such as GitHub and GitLab.
```

> However, there are differences in the details and scope of the features offered.
> To address this situation, the Markdown 

> > used at Github is called **Github-flavored Markdown**

Reviewer: ```suggestion
> used at GitHub is called **GitHub-flavored Markdown**
```

> > and the one used at Gitlab is called **GitLab Flavored Markdown**.

Reviewer: ```suggestion
> and the one used at GitLab is called **GitLab Flavored Markdown**.
```

> - So be aware of

> >   - [Github-flavored Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)

Reviewer: ```suggestion
>   - [GitHub-flavored Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
```
